src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1238,6 +1308,62 @@ static ucs_status_t uct_ib_mlx5dv_check_dc(uct_ib_device_t *dev)
     return status;
 }
 
+static ucs_status_t uct_ib_mlx5dv_check_ud(uct_ib_mlx5_md_t *md)
+{
+    uct_ib_qp_init_attr_t qp_init_attr = {};
+    uct_ib_device_t *dev               = &md->super.dev;
+    ucs_status_t status;
+    struct ibv_qp *qp;
+    struct ibv_srq *srq;
+    struct ibv_pd *pd;
+    struct ibv_cq *cq;
+
+    status = uct_ib_mlx5dv_create_tmp_resources(dev, &srq, &pd, &cq);
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    qp_init_attr.send_cq             = cq;
+    qp_init_attr.recv_cq             = cq;
+    qp_init_attr.srq                 = srq;
+#if HAVE_DECL_IBV_CREATE_QP_EX
+    qp_init_attr.comp_mask           = IBV_QP_INIT_ATTR_PD;
+    qp_init_attr.pd                  = pd;
+#endif
+    qp_init_attr.qp_type             = IBV_QPT_UD;
+    qp_init_attr.sq_sig_all          = 0;
+    qp_init_attr.cap.max_send_wr     = 128;
+    qp_init_attr.cap.max_recv_wr     = 128;
+    qp_init_attr.cap.max_send_sge    = 2;
+    qp_init_attr.cap.max_recv_sge    = 1;
+    qp_init_attr.cap.max_inline_data = 0;
+
+#if HAVE_DECL_IBV_CREATE_QP_EX
+    qp = UCS_PROFILE_CALL_ALWAYS(ibv_create_qp_ex, dev->ibv_context,
+                                 &qp_init_attr);
+#else
+    qp = UCS_PROFILE_CALL_ALWAYS(ibv_create_qp, pd, &qp_init_attr);
+#endif
+    if (qp == NULL) {
+        ucs_error("md=%p: failed to create %s QP "
+                  "TX wr:%d sge:%d inl:%d RX wr:%d sge:%d: %m",
+                  md, uct_ib_qp_type_str(qp_init_attr.qp_type),
+                  qp_init_attr.cap.max_send_wr, qp_init_attr.cap.max_send_sge,
+                  qp_init_attr.cap.max_inline_data,
+                  qp_init_attr.cap.max_recv_wr, qp_init_attr.cap.max_recv_sge);
+        goto out_close_tmp_resources;
```

Comment:
maybe set status to `NO_MEMORY` error here?

Reply:
fixed, better to use `UCS_ERR_IO_ERROR` as in other places when creating IB objects fail